### PR TITLE
[Metrics][LOG] Add metrics for compaction permits and log for merge rowsets

### PR DESCRIPTION
**Pull request by weizuo93**
## Proposed changes

1. Add metrics to `used permits` and `waitting permits` for compaction. It would be useful to monitor `permits` hold by all executing compaction tasks and waitting compaction task.

2. Add log which can be chosen by config  for merge rowsets.  It would be helpful to track the process of rowsets merging for compaction task which lasts for a long time.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [] I have create an issue on (Fix #ISSUE), and have described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [] I have added tests that prove my fix is effective or that my feature works
- [x] If this change need a document change, I have updated the document
- [x] Any dependent changes have been merged

### CODE REVIEW

File: be/src/olap/merger.cpp [hunk 1]
```diff
@@ -65,6 +65,11 @@ OLAPStatus Merger::merge_rowsets(TabletSharedPtr tablet,
         RETURN_NOT_OK_LOG(dst_rowset_writer->add_row(row_cursor),
                           "failed to write row when merging rowsets of tablet " + tablet->full_name());
         output_rows++;
+        if (config::row_step_for_compaction_merge_log != 0 && output_rows % config::row_step_for_compaction_merge_log == 0) {
```
Comment (morningman) [inline be/src/olap/merger.cpp:68]: There is "LOG_EVERY_N" macro for this purpose

Comment (weizuo93) [reply be/src/olap/merger.cpp:68]: > There is "LOG_EVERY_N" macro for this purpose

OK.

Comment (weizuo93) [reply be/src/olap/merger.cpp:68]: > There is "LOG_EVERY_N" macro for this purpose

"LOG_EVERY_N" will print a message after "LOG_EVERY_N" is executed `n` times. After test，"LOG_EVERY_N" macro is for all compaction threads，it means  a message will be printed after `n` rows merged for all compaction tasks. But we need a message after `n` rows merged for a particular compaction task.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**